File: DragonsGenerator.API/Models/Language.cs
namespace DragonsGenerator.API.Models;

public record Language(
    string Id,
    string Name,
    string Category,
    LanguageLinguistics Linguistics,
    LanguageSpeakers Speakers,
    LanguageLore Lore
);

public record LanguageLinguistics(
    List<LanguageWritingSystem> WritingSystems,
    bool IsOralOnly,
    string? WritingNotes = null
);

public record LanguageWritingSystem(
    string Id,
    string Label,
    string Type
);

public record LanguageSpeakers(
    List<SpeakerRef> Primary,
    List<string> Regions,
    bool? IsExtinct = null
);

public record SpeakerRef(string Id, string Label);

public record LanguageLore(
    string FullDescription,
    string? Sonority = null
);

File: DragonsGenerator.WEB/src/app/core/models/Languages/language.ts
export interface Language {
  id: string;
  name: string;
  category: string; // ex: 'base', 'exotique', 'secret'
  linguistics: LanguageLinguistics;
  speakers: LanguageSpeakers;
  lore: LanguageLore;
}

export interface LanguageLinguistics {
  writingSystems: WritingSystem[];
  isOralOnly: boolean;
  writingNotes?: string | null;
}

export interface WritingSystem {
  id: string;
  label: string;
  type: string;
}

export interface LanguageSpeakers {
  primary: SpeakerRef[];
  regions: string[];
  isExtinct?: boolean | null;
}

export interface SpeakerRef {
  id: string;
  label: string;
}

export interface LanguageLore {
  fullDescription: string;
  sonority?: string | null;
}
